src/main/kotlin/com/nikitanazarov/coroutinestacks/CoroutineStacksPanel.kt

```diff
@@ -1,64 +1,145 @@
 package com.nikitanazarov.coroutinestacks
 
-import com.intellij.debugger.engine.DebugProcessListener
-import com.intellij.debugger.engine.SuspendContext
+import com.intellij.debugger.engine.*
+import com.intellij.debugger.engine.events.DebuggerCommandImpl
 import com.intellij.debugger.impl.DebuggerManagerListener
 import com.intellij.debugger.impl.DebuggerSession
+import com.intellij.openapi.application.runInEdt
 import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.ComboBox
 import com.intellij.ui.components.JBPanelWithEmptyText
-import com.mxgraph.swing.mxGraphComponent
-import com.mxgraph.view.mxGraph
+import com.intellij.xdebugger.XDebuggerManager
+import org.jetbrains.kotlin.idea.debugger.coroutine.data.CoroutineInfoData
+import org.jetbrains.kotlin.idea.debugger.coroutine.data.State
 import org.jetbrains.kotlin.idea.debugger.coroutine.proxy.CoroutineDebugProbesProxy
-import java.awt.BorderLayout
-import javax.swing.JComponent
-import javax.swing.SwingConstants
+import java.awt.*
+import javax.swing.*
 
 class CoroutineStacksPanel(project: Project) : JBPanelWithEmptyText() {
-    private var coroutineGraph: JComponent? = null
+    private val panelContent = Box.createVerticalBox()
+    private val forest = Box.createVerticalBox()
 
     init {
-        layout = BorderLayout()
+        layout = BoxLayout(this, BoxLayout.Y_AXIS)
         emptyText.text = CoroutineStacksBundle.message("no.java.debug.process.is.running")
+        val currentSession = XDebuggerManager.getInstance(project).currentSession
+        if (currentSession != null) {
+            val javaDebugProcess = currentSession.debugProcess as? JavaDebugProcess
+            val currentProcess = javaDebugProcess?.debuggerSession?.process
+            emptyText.component.isVisible = false
```

Comment:
What if current session is not null, but the `javaDebugProcess` variable is null, or `currentProcess` is null? This way we will hide the `No java debug process is currently running` message, when we actually shouldn't have done this. This situation can happen if, for example, a user starts a python debugging session in a Java project.